app/Services/RequestLogService.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\UserRequestLog;
+use App\Models\UserSettings;
+use Closure;
+use Illuminate\Support\Facades\Auth;
+
+class RequestLogService
+{
+
+    public function collectData($request,$response){
+
+        UserRequestLog::create([
+            'user_id' => Auth::user()->id,
+            'url' => $request->url(),
+            'method' => $request->method(),
+            'passed' => '1',
+            'request_body' => $request,
+            'response_body' => $response,
+            'headers' => $request->header()
+        ]);
+
+        return response([
+            'message' => 'data collected successfully',
+        ], 200);
+    }
+
+    public static function checkUser(){
+
+        $user = Auth::user();
+
+        $numOfRequestsToday = $user->UserRequestLog()->where('created_at','>',today())->where('passed',1)->count();
+
+        $numOfRequestsToday >= $user->UserSetting()->max_requests ? $permission = false : $permission = true;
+
+        return $permission;
```

Comment:
you can do 

return ($numOfRequestsToday >= $user->UserSetting()->max_requests);

Reply:
To make it more clear that you are returning a boolean you can add return !!(condition) 

! - ხო უარყოფაა
!! - უარყოფის უარყოფაა- იგივე მნიშვნელობას დატოვებს უბრალოდ აშკარა გახდება და მარტივი დასანახი რომ ბულიანია

Reply:
> To make it more clear that you are returning a boolean you can add return !!(condition)
> 
> ! - ხო უარყოფაა !! - უარყოფის უარყოფაა- იგივე მნიშვნელობას დატოვებს უბრალოდ აშკარა გახდება და მარტივი დასანახი რომ ბულიანია

თუ ჩემი true მნიშვნელობა << ! >> ოპერატორის ის შემთხვევაში false მნიშვნელობას იღებს << ! ! >> ს შემთხვევაში ხო true იქნება 
მაშინ ფერმიშენებში უნდა შევცვალო მნიშვნელობები და იქ წაკითხვა უფრო ისეთი არ იქნება ?

Reply:
მარტივად დასანახად შეიძლება მსგავსი რამე გაკეთდეს ?
![image](https://github.com/AleksandreGaspariani/stackexchange_api/assets/87417497/7cbef5f8-cc24-4b4b-9b3b-894aeb0d42b2)

Reply:
> მარტივად დასანახად შეიძლება მსგავსი რამე გაკეთდეს ?
> 
> ![image](https://github.com/AleksandreGaspariani/stackexchange_api/assets/87417497/7cbef5f8-cc24-4b4b-9b3b-894aeb0d42b2)
> 
> 
> 
> 

თუ ეგეთ მეთოდს აკეთებ უფრო აზრიანი სახელით და პირიქით უნდა იყოს მაშინ, იუზერ სეთინგი კი არ უნდა ამოწმებდეს იუზერს არამედ იუზერს უნდა ჰქონდეს მეთოდი User::hasRequests() ან რამე მსგავსი და მეთოდის სახელიც უნდა ისეთი იყოს რო აშკარად ბულს აბრუნებს პასუხად

Reply:
თორე ესე checkUser ი რას ჩეკავს არ არის აშკარა

Reply:
> > To make it more clear that you are returning a boolean you can add return !!(condition)
> > ! - ხო უარყოფაა !! - უარყოფის უარყოფაა- იგივე მნიშვნელობას დატოვებს უბრალოდ აშკარა გახდება და მარტივი დასანახი რომ ბულიანია
> 
> თუ ჩემი true მნიშვნელობა << ! >> ოპერატორის ის შემთხვევაში false მნიშვნელობას იღებს << ! ! >> ს შემთხვევაში ხო true იქნება მაშინ ფერმიშენებში უნდა შევცვალო მნიშვნელობები და იქ წაკითხვა უფრო ისეთი არ იქნება ?

რაიო ?

Reply:
> > To make it more clear that you are returning a boolean you can add return !!(condition)
> > ! - ხო უარყოფაა !! - უარყოფის უარყოფაა- იგივე მნიშვნელობას დატოვებს უბრალოდ აშკარა გახდება და მარტივი დასანახი რომ ბულიანია
> 
> თუ ჩემი true მნიშვნელობა << ! >> ოპერატორის ის შემთხვევაში false მნიშვნელობას იღებს << ! ! >> ს შემთხვევაში ხო true იქნება


ეს მსჯელობა აქ მთავრდება, მერე რა პერმიშენზე ლაპარაკობ ?